src/x_config_gen/device_section.rs

```diff
@@ -0,0 +1,207 @@
+use std::fmt::{self};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum DeviceSectionElement {
```

Comment:
if you plan on adding values there then it might be good to mark it as [non_exhaustive](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute)